src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -75,8 +90,34 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
         public async Task<IPackageSearchMetadata> GetLatestPackageMetadataAsync(PackageIdentity identity,
             bool includePrerelease, CancellationToken cancellationToken)
         {
+            VersionRange allowedVersions = null;
+
+            // at this time, allowed version range should be applied at project level. 
+            // For solution level, we will apply allowed version range for all the selected projects in detail control model
+            if (!_isSolution)
+            {
+                if (_packageReferences == null)
+                {
+                    // get all package references for all the projects and cache locally
+                    var packageReferencesTasks = _projects.Select(project => project.GetInstalledPackagesAsync(cancellationToken));
+                    _packageReferences = (await Task.WhenAll(packageReferencesTasks)).SelectMany(p => p).Where(p => p != null);
```

Comment:
yes, first time packageReferences will be null. And it will be initialized which will be cached and reused.